### PR TITLE
[ERE-1825] Manage User Change Email Address

### PR DESCRIPTION
**Pull request by ppettitau**

### CODE REVIEW

File: rdrf/rdrf/users/views.py [hunk 1]
```diff
@@ -86,3 +155,5 @@ def post(self, request, *args, **kwargs):
         if request.user == self.user:
             auth.logout(request)  # Force logout, requiring the user to reauthenticate with their new email address
```
Comment (ppettitau) [inline rdrf/rdrf/users/views.py:153]: Wondering if I should do this instead: https://github.com/django/django/blob/107865780aa44914e21d27fdf4ca269bc61c7f01/django/contrib/auth/views.py#L369

File: rdrf/rdrf/users/utils.py [hunk 2]
```diff
@@ -84,11 +85,16 @@ def _send_email_change_request_completed_notification(user, user_previous_email)
     email_template = get_template('registration/email_reset_completed.html')
     email_recipient = {user_previous_email: user.preferred_language}
 
+    if user.my_registry:
```
Comment (ScriptSmith) [inline rdrf/rdrf/users/utils.py]: Can use the walrus operator to save querying twice:

```python
if registry := user.my_registry:
    registry_name = registry.name
else:
    registry_name = settings.PROJECT_TITLE
```

File: rdrf/rdrf/users/views.py [hunk 3]
```diff
@@ -4,132 +4,154 @@
...
+        return user  # Superuser has automatic access to this user
 
-        context = {'registry': self.user.my_registry,
-                   'pending_request': pending_request}
+    if request_user.is_staff:
+        # Filter the users by those that the current logged in staff member has access to
+        users = CustomUser.objects \
+            .filter(Q(working_groups__in=request_user.working_groups.all()) | Q(working_groups__isnull=True)) \
+            .filter(registry__in=request_user.registry.all()) \
+            .filter(is_superuser=False)
```
Comment (ScriptSmith) [inline rdrf/rdrf/users/views.py]: The query in `CustomUserAdmin` should use this code through a common function. May still need add the `distinct` there (but not necessarily here).

**Review by ScriptSmith — APPROVED** [review]

File: rdrf/rdrf/users/views.py (hunk 1 above)
Comment (ScriptSmith) [reply rdrf/rdrf/users/views.py:153]: Yeah, seems to be recommended in the docs:

https://docs.djangoproject.com/en/3.2/topics/auth/default/#session-invalidation-on-password-change

Comment (ppettitau) [reply rdrf/rdrf/users/views.py:153]: This only works if the password value has changed.
https://docs.djangoproject.com/en/2.2/topics/auth/customizing/#django.contrib.auth.models.AbstractBaseUser.get_session_auth_hash

File: rdrf/rdrf/users/utils.py (hunk 2 above)
Comment (ppettitau) [reply rdrf/rdrf/users/utils.py]: Done.

File: rdrf/rdrf/users/views.py (hunk 3 above)
Comment (ppettitau) [reply rdrf/rdrf/users/views.py]: moved to CustomUser object manager

**Pull request merged**